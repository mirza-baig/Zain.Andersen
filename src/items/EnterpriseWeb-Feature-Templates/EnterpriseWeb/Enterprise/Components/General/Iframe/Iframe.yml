---
ID: "79817669-c516-4474-8a1e-0391ee5ef169"
Parent: "174d3455-3c7f-4c47-9b10-196fafc37558"
Template: "ab86861a-6030-46c5-b394-e8f99e8b87db"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/General/Iframe/Iframe
SharedFields:
- ID: "06d5295c-ed2f-4a54-9bf2-26228d113318"
  Hint: __Icon
  Value: Control/32x32/group_h.png
- ID: "12c33f3f-86c5-43a5-aeb4-5598cec45116"
  Hint: __Base template
  Value: |
    {B61A1D20-9EE5-4CCB-9E0D-24A97E58D293}
    {D47D8920-5527-4C26-92E8-BD999720B197}
- ID: "f7d48a55-2158-4f02-9356-756654404f73"
  Hint: __Standard values
  Value: "{2902D36F-C5C1-427B-8025-276EE944C2D6}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20230228T234859Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "9e95b39c-5fa6-4c8d-bbc6-39a39a0ed212"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230420T220037Z
