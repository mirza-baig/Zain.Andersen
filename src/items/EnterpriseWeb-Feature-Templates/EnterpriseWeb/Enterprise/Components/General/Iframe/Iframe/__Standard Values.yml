---
ID: "2902d36f-c5c1-427b-8025-276ee944c2d6"
Parent: "79817669-c516-4474-8a1e-0391ee5ef169"
Template: "79817669-c516-4474-8a1e-0391ee5ef169"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/General/Iframe/Iframe/__Standard Values
SharedFields:
- ID: "b03569b1-1534-43f2-8c83-bd064b7d782c"
  Hint: __Renderers
  Value: "{9D5851A0-C782-462B-AF45-5C112A998A06}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20230301T201200Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "07077465-3c33-4295-bb2b-2cb4ab70421e"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230301T214608Z
