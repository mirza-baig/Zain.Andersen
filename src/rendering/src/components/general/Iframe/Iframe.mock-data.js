const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'Iframe',
  },
  fields: {
    "iframeUrl": {
      "value": "https://techpub1.andersenwindows.com/?site=AW"
    },
    "iframeAttributes": {
      "value": "loading=lazy&iframeborder=0&allowfullscreen=true&scrolling=no"
    },
    "iframeResizerOptions": {
      "value": "log=true"
    },
    "sectionId": {
      "value": ""
    },
    "componentSpacing": {
      "id": "c63ff0fc-51d0-43d3-be0f-1884b3e98564",
      "url": "http://cm/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Component-Spacing/Padding-Standard",
      "name": "Padding Standard",
      "displayName": "Padding Standard",
      "fields": {
        "Value": {
          "value": "paddingStandard"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "backgroundColor": {
      "id": "6c8aa36e-f22d-400a-9fe7-391181db23ad",
      "url": "http://cm/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Background-Color/Black",
      "name": "Black",
      "displayName": "Black",
      "fields": {
        "Value": {
          "value": "black"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
