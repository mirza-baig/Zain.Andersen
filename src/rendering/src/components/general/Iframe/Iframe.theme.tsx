// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const IframeTheme = (): ThemeFile => {
  return {
    aw: {
      classes: {
        iframeContainerClass: 'col-span-12 w-full lg:mx-auto md:max-w-screen-lg px-m',
      },
    },
    rba: {
      classes: {
        iframeContainerClass: 'col-span-12 w-full lg:mx-auto md:max-w-screen-lg px-m',
      },
    },
  };
};
