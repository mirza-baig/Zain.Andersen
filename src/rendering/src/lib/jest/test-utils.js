// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
