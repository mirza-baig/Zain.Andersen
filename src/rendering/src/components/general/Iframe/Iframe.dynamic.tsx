// Global
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import IframeResizer from 'iframe-resizer-react';
import { useTheme } from 'lib/context/ThemeContext';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import { getEnum } from 'lib/utils';
import { Component } from 'src/helpers/Component';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';
import { IframeTheme } from './Iframe.theme';

export type IframeProps = Feature.EnterpriseWeb.Enterprise.Components.General.Iframe.Iframe;

const ParseNameValuePairs = (source: string): Array<unknown> => {
  const returnValue: Array<unknown> = [];

  const parameters = source?.split('&');

  if (parameters?.length > 0) {
    parameters.forEach(function (x) {
      let splitX = [];
      splitX = x.split('=');

      if (splitX?.length > 1) {
        const obj: Record<string, string> = {};

        obj[splitX[0]] = splitX[1];
        returnValue.push([splitX[0], splitX[1]]);
      }
    });
  }
  return returnValue;
};
/* eslint-disable @typescript-eslint/no-explicit-any */
const GetIframeHtml = (parameters: Array<any>): JSX.Element => {
  const iframeResizerProps: { [index: string]: any } = {};

  if (parameters) {
    parameters.forEach((x) => x);
    for (const i in parameters) {
      const currentParameter = parameters[i];
      const currentParameterKey = currentParameter[0] as string;
      const currentParameterValue = currentParameter[1] as string;
      if (iframeResizerProps[currentParameterKey] === undefined) {
        iframeResizerProps[currentParameterKey] = currentParameterValue;
      }
    }
  }

  const iFrameElement = IframeResizer(iframeResizerProps);

  return <>{iFrameElement}</>;
};

const Iframe = (props: IframeProps) => {
  const { themeData } = useTheme(IframeTheme());
  const style = getEnum<ComponentBackgroundVariants>(props.fields?.backgroundColor) ?? 'white';

  const resizerOptions = ParseNameValuePairs(props?.fields?.iframeResizerOptions?.value);
  const iframeAttributes = ParseNameValuePairs(props?.fields?.iframeAttributes?.value);
  let finalIframeAttributes: Array<any> = [
    ['className', themeData.classes.iframeContainerClass],
    ['src', props?.fields?.iframeUrl?.value],
  ];

  if (props?.fields?.iframeTitle?.value) {
    finalIframeAttributes.push(['title', props?.fields?.iframeTitle?.value]);
  }

  try {
    if (resizerOptions?.length > 0) {
      finalIframeAttributes = finalIframeAttributes.concat(resizerOptions);
    }

    if (iframeAttributes?.length > 0) {
      finalIframeAttributes = finalIframeAttributes.concat(iframeAttributes);
    }
  } catch {}

  return (
    <Component
      variant="full"
      backgroundVariant={style}
      padding=" "
      dataComponent="general/iframe"
      {...props}
    >
      {GetIframeHtml(finalIframeAttributes)}
    </Component>
  );
};

export default withDatasourceCheck()<IframeProps>(Iframe);
