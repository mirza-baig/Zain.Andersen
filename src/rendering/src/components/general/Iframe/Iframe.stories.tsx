// Global
import { Story, Meta } from '@storybook/react';
// Local
import Iframe, { IframeProps } from './Iframe.dynamic';
import defaultData from './Iframe.mock-data';

export default {
  title: 'General/Iframe',
  component: Iframe,
} as Meta;
const Template: Story<IframeProps> = (props) => <Iframe {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
