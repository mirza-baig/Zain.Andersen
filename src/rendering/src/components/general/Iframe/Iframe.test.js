// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import Iframe from './Iframe.dynamic';
import defaultData from './Iframe.mock-data';

it('renders correctly', async () => {
  const component = snapshot(Iframe, { componentProps: defaultData });
  await hasDataComponent(component, 'general/iframe');
});
