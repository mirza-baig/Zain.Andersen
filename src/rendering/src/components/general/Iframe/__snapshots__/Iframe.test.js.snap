// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/iframe"
  id="id1695242607"
>
  <div
    class="py-8 w-full theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-s md:gap-s "
    >
      <iframe
        allowfullscreen="true"
        class="col-span-12 w-full lg:mx-auto md:max-w-screen-lg px-m"
        id="iFrameResizer0"
        iframeborder="0"
        loading="lazy"
        scrolling="no"
        src="https://techpub1.andersenwindows.com/?site=AW"
        style="overflow: auto;"
      />
    </div>
  </div>
</section>
`;
